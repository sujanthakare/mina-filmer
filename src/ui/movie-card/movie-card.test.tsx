import { fireEvent, render, screen } from '@testing-library/react';

import MovieCard from './movie-card';

describe('MovieCard', () => {
  it('should call onFavoriteClick if click on mark favorite button', () => {
    const mocked_onFavoriteClick = jest.fn();
    render(
      <MovieCard
        onFavoriteClick={mocked_onFavoriteClick}
        movie={{
          id: 1,
          title: 'TITLE',
          poster_path: 'POST_PATH',
        }}
      />
    );

    const favoriteButton = screen.getByTestId('favorite-button-1');

    fireEvent.click(favoriteButton);

    expect(mocked_onFavoriteClick).toBeCalledTimes(1);
  });

  it('should call onWatchLaterClick if click on mark watch later button', () => {
    const mocked_onWatchLaterClick = jest.fn();
    render(
      <MovieCard
        onWatchLaterClick={mocked_onWatchLaterClick}
        movie={{
          id: 1,
          title: 'TITLE',
          poster_path: 'POST_PATH',
        }}
      />
    );

    const watchLaterButton = screen.getByTestId('watch-later-button-1');

    fireEvent.click(watchLaterButton);

    expect(mocked_onWatchLaterClick).toBeCalledTimes(1);
  });

  it('should match snapshot with movie', () => {
    const { container } = render(
      <MovieCard
        movie={{
          id: 1,
          title: 'TITLE',
          poster_path: 'POST_PATH',
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot if movie is marked as favorite and watch later', () => {
    const { container } = render(
      <MovieCard
        favorite
        watchLater
        movie={{
          id: 1,
          title: 'TITLE',
          poster_path: 'POST_PATH',
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
