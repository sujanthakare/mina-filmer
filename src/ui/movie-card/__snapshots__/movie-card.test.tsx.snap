// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match snapshot if movie is marked as favorite and watch later 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-testid="movie-card"
  >
    <img
      alt="TITLE"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      src="https://image.tmdb.org/t/p/original/POST_PATH"
    />
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
    >
      <button
        aria-label="Mark favorite"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-testid="favorite-button-1"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
          data-testid="FavoriteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Mark watch later"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-testid="watch-later-button-1"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
          data-testid="WatchLaterIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm4.2 14.2L11 13V7h1.5v5.2l4.5 2.7-.8 1.3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`MovieCard should match snapshot with movie 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-testid="movie-card"
  >
    <img
      alt="TITLE"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      src="https://image.tmdb.org/t/p/original/POST_PATH"
    />
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
    >
      <button
        aria-label="Mark favorite"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-testid="favorite-button-1"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FavoriteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Mark watch later"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-testid="watch-later-button-1"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="WatchLaterIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm4.2 14.2L11 13V7h1.5v5.2l4.5 2.7-.8 1.3z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
